### PR TITLE
feat: improved translation

### PR DESCRIPTION
**Pull request by Henvy-Mango**
# Tittle Here

improved translation
## Description

add deepL translator
add yandex translator
## Check List

- [x] I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] My code follows the code style of this project
- [x] I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: TMessagesProj/src/main/java/top/qwq2333/nullgram/translate/BaseTranslator.kt [hunk 1]
```diff
@@ -62,11 +62,12 @@ abstract class BaseTranslator {
                 ignoreUnknownKeys = true
             })
         }
+        install(HttpCookies)
         // Ensure using UTF-8
         Charsets {
             register(Charsets.UTF_8)
             sendCharset = Charsets.UTF_8
-            responseCharsetFallback = Charset.forName("GBK")
+            responseCharsetFallback = Charsets.UTF_8
```
Comment (qwq233) [inline TMessagesProj/src/main/java/top/qwq2333/nullgram/translate/BaseTranslator.kt:70]: fallback为什么要改

File: TMessagesProj/src/main/java/top/qwq2333/nullgram/helpers/TranslateHelper.kt [hunk 2]
```diff
@@ -93,8 +94,9 @@ object TranslateHelper {
         YandexTranslator(2),
         TelegramTranslator(3),
         MicrosoftTranslator(4),
-        LingoTranslator(5),
-        BaiduTranslator(6),
+        DeepLTranslator(5),
```
Comment (qwq233) [inline TMessagesProj/src/main/java/top/qwq2333/nullgram/helpers/TranslateHelper.kt:97]: 不要改变已有的num

File: TMessagesProj/src/main/java/top/qwq2333/nullgram/activity/GeneralSettingActivity.java [hunk 3]
```diff
@@ -240,13 +243,23 @@ protected void onItemClick(View view, int position, float x, float y) {
...
             TranslateHelper.showTranslationProviderSelector(getParentActivity(), view, null, param -> {
                 if (param) {
                     listAdapter.notifyItemChanged(translationProviderRow, PARTIAL);
                 } else {
                     listAdapter.notifyItemChanged(translationProviderRow, PARTIAL);
                     listAdapter.notifyItemChanged(translationTargetRow, PARTIAL);
                 }
+                if (!oldProvider.equals(TranslateHelper.getCurrentProviderType())) {
+                    if (oldProvider.equals(ProviderType.DeepLTranslator)) {
+                        listAdapter.notifyItemRemoved(deepLFormalityRow);
```
Comment (qwq233) [inline TMessagesProj/src/main/java/top/qwq2333/nullgram/activity/GeneralSettingActivity.java:256]: 这堆判断放update row方法 判断一次就够了 不需要额外remove那些

File: TMessagesProj/src/main/java/top/qwq2333/nullgram/translate/providers/DeepLTranslator.kt [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+                else -> {
+                    Log.w(it.bodyAsText())
+                    return RequestResult(from, null, it.status)
+                }
+            }
+        }
+    }
+
+    const val FORMALITY_DEFAULT = 0
```
Comment (qwq233) [inline TMessagesProj/src/main/java/top/qwq2333/nullgram/translate/providers/DeepLTranslator.kt:116]: 这里应用data class 和 序列化 可以参考 nicegram controller

**Review by qwq233 — CHANGES_REQUESTED** [review]

**Review by qwq233 — DISMISSED** [review]

**Comment by qwq233** [comment]
The base branch requires all commits to be signed.

**Comment by Henvy-Mango** [comment]
> The base branch requires all commits to be signed.

好了

**Review by qwq233 — APPROVED** [review]

**Pull request merged**